src/Collection/PromiseCollection.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace MyParcelCom\ApiSdk\Collection;
+
+use MyParcelCom\ApiSdk\Resources\Interfaces\ResourceInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class PromiseCollection implements CollectionInterface
+{
+    /** @var callable */
+    protected $promiseCreator;
+
+    /** @var callable */
+    protected $resourceCreator;
+
+    /** @var ResourceInterface[] */
+    protected $resources = [];
+
+    /** @var int */
+    protected $offset = 0;
+
+    /** @var int */
+    protected $limit = 30;
+
+    /** @var int */
+    protected $count;
+
+    /** @var int */
+    protected $currentResourceNumber = 0;
+
+    /**
+     * PromiseCollection constructor.
+     *
+     * @param callable $promiseCreator
+     * @param callable $resourceCreator
+     */
+    public function __construct(callable $promiseCreator, callable $resourceCreator)
+    {
+        $this->promiseCreator = $promiseCreator;
+        $this->resourceCreator = $resourceCreator;
+    }
+
+    /**
+     * Counts the amount of resources in the collection.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        if (!isset ($this->count)) {
+            $this->get();
+        }
+
+        return $this->count;
+    }
+
+    /**
+     * Retrieves the resources based on limit and offset.
+     * Max (and default) limit is 30.
+     *
+     * @return ResourceInterface[]
+     */
+    public function get()
+    {
+        if (isset($this->count) && $this->offset >= $this->count) {
+            return [];
+        }
+
+        $firstPage = ceil(($this->offset + 1) / 30);
```

Comment:
You should add a comment of what's going on here. I can imagine that someone looking at this code has no idea what the 2 pages are.